docs/adr/0006-server-arch.md

```diff
@@ -0,0 +1,185 @@
+# 0006. Server Architecture
+
+Date: 7th June 2019
+
+## Status
+
+Proposed
+
+## Context
+
+The server-side code has a number of issues with it which make it difficult to
```

Comment:
Empirically, if people find it difficult to work with, it means it's difficult to work with. I wouldn't overdiagnose it as SOLID is specific to object-oriented programming in languages that tend towards static typing.

Reply:
Fair point - I guess the reference to SOLID here should be a collection of good practices for the readers reference.